src/openstudio_lib/ThermalZonesGridView.cpp

```diff
@@ -383,35 +383,32 @@ void ThermalZonesGridController::addColumns(const QString& /*category*/, std::ve
       });
 
       std::function<bool(model::ThermalZone*, const model::Schedule&)> setCoolingSchedule([](model::ThermalZone* z, model::Schedule t_s) {
+        bool result = false;
         if (boost::optional<model::ThermostatSetpointDualSetpoint> thermostat = z->thermostatSetpointDualSetpoint()) {
-          return thermostat->setCoolingSetpointTemperatureSchedule(t_s);
+          result = thermostat->setCoolingSetpointTemperatureSchedule(t_s);
         } else {
           model::ThermostatSetpointDualSetpoint t(z->model());
+          result = t.setCoolingSetpointTemperatureSchedule(t_s);
           z->setThermostatSetpointDualSetpoint(t);
-          return t.setCoolingSetpointTemperatureSchedule(t_s);
         }
+        return result;
       });
 
       boost::optional<std::function<void(model::ThermalZone*)>> resetCoolingSchedule([](model::ThermalZone* z) {
         if (boost::optional<model::ThermostatSetpointDualSetpoint> thermostat = z->thermostatSetpointDualSetpoint()) {
           thermostat->resetCoolingSetpointTemperatureSchedule();
+          if (!thermostat->heatingSetpointTemperatureSchedule()) {
+            z->resetThermostatSetpointDualSetpoint();
+          }
```

Comment:
ok, if you remove the cooling schedule and no heating schedule was assigned, you reset the ThermostatSetpointDualSetpoint.